File: Ruby/problem_7.rb
def is_prime(num)
  divisor = 2
  while (divisor * divisor <= num)
    if num % divisor == 0
      return false
    else
      divisor += 1
    end
  end
  true
end

count = 1
num = 2
while (count <= 10001)
  if is_prime(num)
    puts num
    count += 1
  end
  num += 1
end

File: JS/problem7.js
var isPrime = function(num) {
  var divisor = 2;
  for (; divisor*divisor <= num; divisor++) {
    if (num % divisor === 0) {
      return false;
    }
  }
  return true;
};

var count = 0,
    num   = 2;

do {
  if (isPrime(num)) {
    count++;
    console.log("prime # " + count + " is: " + num);
  }
  num++;
} while (count <= 10000);
